bin/link_changelog

```diff
@@ -9,30 +9,49 @@ set -euo pipefail
 #
 # Usage:
 #
-# $ link_changelog <package_name> <git_tag>
+# $ link_changelog <git_tag>
 #
 # For example:
 #
 # $ link_changelog some_single_package v0.7.3
 # 
-# $ link_changelog leancode_contracts leancode_contracts-v2.1.0
-
-package_name="${1:-}"
-tag="${2:-}"
+# $ link_changelog leancode_contracts v2.1.0
+#
+# $ link_changelog dispose_scope-v3.0.0-dev.1
 
-if [ -z "$package_name" ]; then
-    echo "error: missing package name" 1>&2
-    exit 1
+if [ "$#" = 1 ]; then
+    tag="${1:-}"
+elif [ "$#" = 2 ]; then
+    package_name="${1:-}"
+    tag="${2:-}"
 fi
 
 if [ -z "$tag" ]; then
     echo "error: missing tag" 1>&2
     exit 1
 fi
 
-prefix="${3:-${package_name}-}"
-version="${tag#$prefix}"
-version="${version#v}"
-version=$(echo "$version" | tr -d '.+')
-link="https://pub.dev/packages/$package_name/changelog#$version"
-echo "$link"
+if [ -n "${package_name:-}" ]; then
+    # package name + git tag
+    version="${tag#"$package_name"}"
+    version="${version#v}"
+    version=$(echo "$version" | tr -d '.+')
+    link="https://pub.dev/packages/$package_name/changelog#$version"
+    echo "$link"
+else
+    # only git tag
+    package_name="$(echo "$tag" | cut -d '-' -f 1)"
+    version="$(echo "$tag" | cut -d '-' -f 2)"
```

Comment:
Won't the sample from line 20, `link_changelog dispose_scope-v3.0.0-dev.1`, be cut… strangely by the second line here? I think it will result in `-dev.1` being silently cut off. Or is it intended behavior?

Also you could probably simplify calls to `cut` by using builtin string chopping operators, assuming that you indeed want to keep everything after first `-` in `version`:
```suggestion
    package_name="${tag%%-*}"
    version="${tag#*-}"
```

Reply:
Thanks. Cutting git tag suffixes like `-dev.1` was indeed a bug. I fixed it.

Also, I really appreciate and admire you using these building chopping operators, but TBH, I prefer a simple `cut` – it's just more readable, and I really don't care about starting a subprocess to do such a simple thing. 😛